crates/web5/src/credentials/verifiable_credential_1_1.rs

```diff
@@ -205,93 +216,61 @@ impl VerifiableCredential {
 
         let kid = header
             .claim("kid")
             .and_then(serde_json::Value::as_str)
-            .ok_or_else(|| JosekitError::InvalidJwtFormat(CredentialError::MissingKid.into()))?
-            .to_string();
-
-        let did = Did::parse(&kid)?;
+            .ok_or(CredentialError::MissingKid)?;
 
-        let resolution_result = ResolutionResult::resolve(&did.uri);
-        if let Some(err) = resolution_result.resolution_metadata.error.clone() {
-            return Err(CredentialError::Resolution(err));
+        if kid.is_empty() {
+            return Err(CredentialError::MissingKid.into());
         }
 
-        let public_key_jwk = resolution_result
-            .document
-            .ok_or_else(|| {
-                JosekitError::InvalidJwtFormat(ResolutionMetadataError::InternalError.into())
-            })?
-            .find_verification_method(FindVerificationMethodOptions {
-                verification_method_id: Some(kid.to_string()),
-            })?
-            .public_key_jwk;
-
-        let verifier = Ed25519Verifier::new(public_key_jwk);
-
-        Self::verify_with_verifier(vc_jwt, Arc::new(verifier))
-    }
-
-    pub fn verify_with_verifier(vc_jwt: &str, verifier: Arc<dyn Verifier>) -> ResultOld<Self> {
-        let header = josekit::jwt::decode_header(vc_jwt)?;
-
-        let kid = header
-            .claim("kid")
-            .and_then(serde_json::Value::as_str)
-            .ok_or_else(|| JosekitError::InvalidJwtFormat(CredentialError::MissingKid.into()))?
-            .to_string();
+        let jwt_payload = match verify {
+            true => {
+                let did = Did::parse(kid)?;
 
-        let jose_verifier = &JoseVerifier { kid, verifier };
+                let resolution_result = ResolutionResult::resolve(&did.uri);
+                if let Some(err) = resolution_result.resolution_metadata.error.clone() {
+                    return Err(err.into());
+                }
 
-        let (jwt_payload, _) = josekit::jwt::decode_with_verifier(vc_jwt, jose_verifier)?;
+                let public_key_jwk = resolution_result
+                    .document
+                    .ok_or(ResolutionMetadataError::InternalError)?
+                    .find_verification_method(FindVerificationMethodOptions {
+                        verification_method_id: Some(kid.to_string()),
+                    })?
+                    .public_key_jwk;
+
+                let jose_verifier = &JoseVerifier {
+                    kid: kid.to_string(),
+                    verifier: Arc::new(Ed25519Verifier::new(public_key_jwk)),
+                };
+
+                let (jwt_payload, _) = josekit::jwt::decode_with_verifier(vc_jwt, jose_verifier)
+                    .map_err(|e| {
+                        Web5Error::Crypto(format!("vc-jwt failed cryptographic verification {}", e))
+                    })?;
+
+                jwt_payload
+            }
+            false => {
```

Comment:
if statement instead of matching against boolean?

Reply:
done 👍